lib/types.nix

```
@@ -710,6 +710,23 @@ rec {
         in mergedOption.type;
     };
 
+    # A type that is one of several submodules, similiar to types.oneOf but is usable inside attrsOf or listOf
+    # submodules need an option with a type str which is used to find the corresponding type
+    taggedSubmodules =
```

```suggestion
    taggedSubmodule =
```

When used by itself, the option value is just a single submodule.
The naming convention describes the option value rather than the definition syntax (or whether multiple defs are ok).

Also apply to `name` below.